### PR TITLE
(Archiving) Fix behaviour for non-positive limits

### PR DESCRIPTION
**Pull request by islean**
## Description

Currently, the archive-spring-files command in the new archiving flow has trouble handling non-positive integers provided as the limit. This PR addresses this and ensures that tests catch any errors with regards to this.

### Fixed

- A provided limit of a non-positive integer now results in an exit.

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b archiving-fix-archive-zero -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: tests/meta/archive/test_archive_api.py [hunk 1]
```diff
@@ -148,6 +147,7 @@ def test_call_corresponding_archiving_method(spring_archive_api: SpringArchiveAP
     mock_request_submitter.assert_called_once_with(files_and_samples=[file_and_sample])
 
 
+@pytest.mark.parametrize("limit", [None, -1, 0, 1])
```
Comment (Vince-janv) [inline tests/meta/archive/test_archive_api.py:150]: 🤩

**Review by Vince-janv — APPROVED** [review]
Good patch 👍

**Comment by islean** [comment]
Currently on master:

Behaviour for limit=0 still archives:
```
θ62° 1d [isak.ohlsson@hasta:~] [S_base] 22s $ cg archive archive-spring-files --limit 0
/home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages/urllib3/connectionpool.py:1095: InsecureRequestWarning: Unverified HTTPS request is being made to host 'cg-df-cat.scilifelab.se'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  warnings.warn(
Sending request with headers: 
{'Content-Type': 'application/json', 'accept': 'application/json', 'Authorization': 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNzAxODY3ODMzLCJqdGkiOiI1OTFhYjYwNzAyNTQ0YmUxYmUwNTY2N2I4OTM5OGFkMiIsInVzZXJfaWQiOjQsImlhdCI6MTcwMTg2NzUzMywic3ViIjoic2NsdXNlciIsImRiX25hbWUiOiJBREEiLCJsZWdhY3lfdG9rZW4iOiJleUpoYkdjaU9pSklVekkxTmlKOS5leUpxZEdraU9qRTNNeXdpYVdGMElqb3hOekF4T0RZM05UTXpMQ0p6ZFdJaU9pSnpZMngxYzJWeUlpd2lhWE56SWpvaVkyY3RaR1l0WTJGMExuTmphV3hwWm1Wc1lXSXVjMlVpZlEublhfR0pBVTlhbFM1SXd3Q0FVRHREeWFHamJUS0JhMDZtOWJIVk1ESkFvYyIsInN1cGVyX3VzZXJfY2xhaW0iOmZhbHNlLCJ1c2VyX2lwIjoiMTMwLjIzNy44MS4xMzciLCJpc3MiOiJwb3NpeCBMaW51eCA0LjE4LjAtNDc3LjI3LjEuZWw4XzgueDg2XzY0IGNnLWRmLWNhdC5zY2lsaWZlbGFiLnNlIn0.NnyVqCbZXYpZFFTFLx5I2hX9iwKtx3NsjfHFY-iOA-A'} 
and body: 
{'osType': 'Unix/MacOS', 'createFolder': True, 'settings': [], 'pathInfo': [{'destination': 'archive@SciLifeLab:/stage/isaks_test_bundle', 'source': 'gpfs@Hasta:/proj/stage/housekeeper-bundles/isaks_test_bundle/2023-12-06/isaks_test_spring_file.spring'}], 'metadataList': []}
/home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages/urllib3/connectionpool.py:1095: InsecureRequestWarning: Unverified HTTPS request is being made to host 'cg-df-cat.scilifelab.se'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  warnings.warn(
Files submitted to karolinska_bucket with archival task id 141.
Adding an archive to file 5048030 with archiving task id 141
θ61° 1d [isak.ohlsson@hasta:~] [S_base] 25s $ 
```
Behaviour for limit=-1 breaks (not too horrible but can be done in a nicer way):
```
sqlalchemy.exc.ProgrammingError: (pymysql.err.ProgrammingError) (1064, "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '-1' at line 7")
[SQL: SELECT file.id AS file_id, file.path AS file_path, file.checksum AS file_checksum, file.to_archive AS file_to_archive, file.version_id AS file_version_id 
FROM file INNER JOIN file_tag_link AS file_tag_link_1 ON file.id = file_tag_link_1.file_id INNER JOIN tag ON tag.id = file_tag_link_1.tag_id INNER JOIN version ON version.id = file.version_id INNER JOIN bundle ON bundle.id = version.bundle_id LEFT OUTER JOIN archive ON file.id = archive.file_id 
WHERE tag.name IN (%(name_1_1)s, %(name_1_2)s) AND (EXISTS (SELECT 1 
FROM archive 
WHERE file.id = archive.file_id)) = false GROUP BY file.id 
HAVING count(tag.name) = %(count_1)s 
 LIMIT %(param_1)s]
[parameters: {'count_1': 2, 'param_1': -1, 'name_1_1': <ArchiveLocations.KAROLINSKA_BUCKET: 'karolinska_bucket'>, 'name_1_2': <SequencingFileTag.SPRING: 'spring'>}]
(Background on this error at: https://sqlalche.me/e/20/f405)
```

**Comment by islean** [comment]
Tests worked fine:

```
1d [isak.ohlsson@hasta:~] [S_base] 4m2s $ cg archive archive-spring-files --limit 0
Please do not provide a non-positive integer as limit - exiting.
1d [isak.ohlsson@hasta:~] [S_base] 27s $ cg archive archive-spring-files --limit -1
Please do not provide a non-positive integer as limit - exiting.
1d [isak.ohlsson@hasta:~] [S_base] 30s $ cg archive archive-spring-files --limit 1
/home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages/urllib3/connectionpool.py:1095: InsecureRequestWarning: Unverified HTTPS request is being made to host 'cg-df-cat.scilifelab.se'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  warnings.warn(
Sending request with headers: 
{'Content-Type': 'application/json', 'accept': 'application/json', 'Authorization': 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNzAxODc2NjMzLCJqdGkiOiIzZjZjNjg5OWM5ZGE0ODcwOTllMGNhOTRiMTU1NDA5ZCIsInVzZXJfaWQiOjQsImlhdCI6MTcwMTg3NjMzMywic3ViIjoic2NsdXNlciIsImRiX25hbWUiOiJBREEiLCJsZWdhY3lfdG9rZW4iOiJleUpoYkdjaU9pSklVekkxTmlKOS5leUpxZEdraU9qRTNOU3dpYVdGMElqb3hOekF4T0RjMk16TXpMQ0pwYzNNaU9pSmpaeTFrWmkxallYUXVjMk5wYkdsbVpXeGhZaTV6WlNJc0luTjFZaUk2SW5OamJIVnpaWElpZlEubnFpNmg0YzFrOHIzMG8tdDNvWTRUM2R2eUVFOFJYYWFpZFJnUzRVc2tmUSIsInN1cGVyX3VzZXJfY2xhaW0iOmZhbHNlLCJ1c2VyX2lwIjoiMTMwLjIzNy44MS4xMzciLCJpc3MiOiJwb3NpeCBMaW51eCA0LjE4LjAtNDc3LjI3LjEuZWw4XzgueDg2XzY0IGNnLWRmLWNhdC5zY2lsaWZlbGFiLnNlIn0.5a-VVi8SfGUT59adGs0vR5Ge9E8zpfn0LatPLdf_wPE'} 
and body: 
{'osType': 'Unix/MacOS', 'createFolder': True, 'settings': [], 'pathInfo': [{'destination': 'archive@SciLifeLab:/stage/isaks_test_bundle', 'source': 'gpfs@Hasta:/proj/stage/housekeeper-bundles/isaks_test_bundle/2023-12-06/isaks_test_spring_file.spring'}], 'metadataList': []}
/home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages/urllib3/connectionpool.py:1095: InsecureRequestWarning: Unverified HTTPS request is being made to host 'cg-df-cat.scilifelab.se'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  warnings.warn(
Files submitted to karolinska_bucket with archival task id 142.
Adding an archive to file 5048030 with archiving task id 142
```

**Comment by islean** [comment]
Deployed to production by IO:

```
done.
Log deploy... done.
cg, version 53.5.2
1d [isak.ohlsson@hasta:~] [P_base] 3m43s $ 
```

**Pull request merged**